File: crowdsale/crowdsale.go
// Copyright 2020 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

package crowdsale

import "github.com/iotaledger/wasp/packages/vm/wasmlib"

func funcInit(ctx wasmlib.ScFuncContext, f *InitContext) {
    if f.Params.Owner().Exists() {
        f.State.Owner().SetValue(f.Params.Owner().Value())
        return
    }
    f.State.Owner().SetValue(ctx.ContractCreator())
}

func funcSetOwner(ctx wasmlib.ScFuncContext, f *SetOwnerContext) {
}

func viewGetOwner(ctx wasmlib.ScViewContext, f *GetOwnerContext) {
}

func funcPurchase(ctx wasmlib.ScFuncContext, f *PurchaseContext) {
}

func funcWithdraw(ctx wasmlib.ScFuncContext, f *WithdrawContext) {
}

func viewPurchaseInfo(ctx wasmlib.ScViewContext, f *PurchaseInfoContext) {
}

func viewPurchaseView(ctx wasmlib.ScViewContext, f *PurchaseViewContext) {
}

File: crowdsale/src/crowdsale.rs
// Copyright 2020 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

use wasmlib::*;

use crate::*;

pub fn func_init(ctx: &ScFuncContext, f: &InitContext) {
    if f.params.owner().exists() {
        f.state.owner().set_value(&f.params.owner().value());
        return;
    }
    f.state.owner().set_value(&ctx.contract_creator());
}

pub fn func_set_owner(_ctx: &ScFuncContext, _f: &SetOwnerContext) {
}

pub fn view_get_owner(_ctx: &ScViewContext, _f: &GetOwnerContext) {
}

pub fn func_purchase(_ctx: &ScFuncContext, _f: &PurchaseContext) {
}

pub fn func_withdraw(_ctx: &ScFuncContext, _f: &WithdrawContext) {
}

pub fn view_purchase_info(_ctx: &ScViewContext, _f: &PurchaseInfoContext) {
}

pub fn view_purchase_view(_ctx: &ScViewContext, _f: &PurchaseViewContext) {
}
